articles/tech-resources/deprecated-extended-error-codes.md

```diff
@@ -0,0 +1,3103 @@
+# Deprecation of API Extended Error Codes
+
+Post January 12th, 2019, SparkPost will no longer return an [extended error code](https://www.sparkpost.com/docs/tech-resources/extended-error-codes/) during for an HTTP response.  This document outlines the deprecation of the extended erroor codes for API responsed in SparkPost.  The reasoning behind this is to standardize our API reponses to provide a more coherent customer experience.  This does not effect extended error codes that are returned via message events in the case of out-of-band errors.
+
+As a general rule, all current error codes will be updated in the following manner:
+1. The description field will be moved to the message field
```

Comment:
How about changing the number list to the following.  I think it'll better outline the changes.  Should have done this before I sent it over

1. The description field will be moved to the message field with the additional changes
    * JSON parsing errors will be returned as:
        > `Request body parsing failed [Attached error message: %error%]`
    * Exists errors will be generic across the following APIs and will return:
        > `resource '%resourcename' already exists`
    * Type errors will no longer return the invalid type, just expected:
        > `'%fieldName' must be a %expectedType%`
    * Require field errors will be generic across the following APIs and will return:
        > `'%fieldName%' is required`
2. The `code` field will be removed
3. The HTTP status code will not change
4. Inline template syntax, compilation, and email_rfc822 validation errors will be moved to out of band errors via message events